src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -204,8 +205,25 @@ public void RegisterCommandHandler<TCommand>(Func<TCommand, KernelInvocationCont
         }
 
         public void RegisterCommandType<TCommand>()
-            where TCommand : KernelCommand =>
-            KernelCommandEnvelope.RegisterCommandTypeReplacingIfNecessary<TCommand>();
+            where TCommand : KernelCommand
+        {
+            if (_registeredCommandTypes.Add(typeof(TCommand)))
+            {
+                KernelCommandEnvelope.RegisterCommandTypeReplacingIfNecessary<TCommand>();
+                var defaultHandler = CreateDefaultHandlerForCommandType<TCommand>();
+                if (defaultHandler is not null)
```

Comment:
It seems like this should not be allowed to return null.

Reply:
done